### PR TITLE
Adds contract events to activity feed.  Closes #123.  Closes #41.

### LINKED ISSUE #41: List of all Notifications
We need a list of all past "Notifications". 
For example:  `Job Accepted by` , `Job Completion Submitted` etc.

### LINKED ISSUE #123: Update Activity Feed with all contract events
Show all contract related events in the activity feed.

Job Posted
Job Started
Job Completed
etc.

### PR DESCRIPTION
**Pull request by deweller**
Use a backend process to sync blockchain events in the background.

Various UI improvements for showing status changes and contract events

All firebase selectors are name-spaced to the current Job contract address

Changes firebase selector for messages and events to jobId/messages and jobId/events

Use environment vars for Firebase

Copy hardhat build artifacts to backend

Adds baseUrl for imports relative to project root

image

Closes #123.  Closes #41.

### CODE REVIEW

**Review by reisr3 — CHANGES_REQUESTED** [review]
Very cool!!! So awesome to have this.

I have two big pieces of feedback:

1. I can't get this to load, I think because my Firebase has different tables - nothing is coming back from Firebase. Can you either programmatically init the tables or provide instructions for how to set this up?
2. The frontend is way too complicated. All that complexity should be handled in the backend. I know we have to deal with the `snapshots` from Firebase, but I think we should just have one table for all these items (we can mix them, just define a `type`). That will help a lot. Then, it should be very simple to filter through the items we want to display per job.

File: frontend/src/components/activity-feed/hooks/useActivityFeed.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+  // load events from firebase
+  const eventsDbRef = job ? contractDatabaseRef(`${job.id}/events`) : null;
+  const [eventSnapshots, eventsAreLoading, eventListError] =
+    useList(eventsDbRef);
+
+  // load messages from firebase
+  const messagesDbRef = job ? contractDatabaseRef(`${job.id}/messages`) : null;
+  const [messageSnapshots, messagesAreLoading, messageListError] =
+    useList(messagesDbRef);
```
[the remark this answers is not in the document]
Comment (deweller) [reply frontend/src/components/activity-feed/hooks/useActivityFeed.ts:20]: I think we need to keep messages and events as separate pieces of data.

The source of truth for Events is the blockchain and need to be synced from the blockchain.  So Events might be deleted and updated if the blockchain re-organizes.  Keeping them in their own key of 'events' makes it simpler to sync on the backend.

The source of truth for Messages is Firebase (for now).  So they are treated differently.  There is no way process that syncs messages to anything else.

I recommend keeping them in a sub-directory named for the jobID.  That way we are only loading into memory the messages for a given jobId.  If all events and messages were at the top level, the web browser client would load ALL messages for all jobs.  That could be a lot of unnecessary data.

In summary, it is intentional that these are separated into jobId/events and jobId/message.

File: frontend/src/components/activity-feed/hooks/activity.d.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { IActivityFeedItem } from 'interfaces/IActivityFeedItem';
```
[the remark this answers is not in the document]
Comment (deweller) [reply frontend/src/components/activity-feed/hooks/activity.d.ts:1]: I put this in a module interfaces folder as it doesn't get used anywhere else in the application.

File: frontend/src/components/activity-feed/hooks/utils/commentActivity.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { ActivityType } from 'enums/ActivityType';
```
[the remark this answers is not in the document]
Comment (deweller) [reply frontend/src/components/activity-feed/hooks/utils/commentActivity.ts:1]: Since these are module-level utilities, I kept this at the module level (in activity-feed) but moved it up one directory for consistency.

File: frontend/src/components/activity-feed/hooks/useActivityFeed.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (reisr3) [reply frontend/src/components/activity-feed/hooks/useActivityFeed.ts:21]: Ok, fine with me for now. Maybe in the future we should do ANOTHER Firebase table that aggregates the two tables...? Like a CQRS thing. I just want to keep the frontend as simple as possible, it should be a dumb UI machine - fed exactly the data it needs.

**Review by reisr3 — APPROVED** [review]
